test/XRP/helpers/xrp-test-utils.ts

```diff
@@ -29,6 +37,76 @@ export default class XRPTestUtils {
     }
     return paymentXrpTransactions
   }
+
+  /**
+   * Generates a random wallet and funds it using the XRPL Testnet faucet.
+   */
+  static async randomWalletFromFaucet(): Promise<Wallet | undefined> {
+    const timeoutInSeconds = 20
+
+    const wallet = Wallet.generateRandomWallet()?.wallet
+    if (!wallet) {
+      throw new XrpError(
+        XrpErrorType.Unknown,
+        'Could not generate a random wallet.',
+      )
+    }
+    const address = wallet.getAddress()
+
+    const classicAddress = XrpUtils.decodeXAddress(address)?.address
+    if (!classicAddress) {
+      throw new XrpError(
+        XrpErrorType.InvalidInput,
+        `Could not decode XAddress ${address}.`,
+      )
+    }
+
+    const rippledUrl = 'test.xrp.xpring.io:50051'
+    const xrpClient = new XrpClient(rippledUrl, XrplNetwork.Test)
+
+    // Balance prior to asking for more funds
+    let startingBalance
+    try {
+      startingBalance = bigInt(await xrpClient.getBalance(address))
+    } catch {
+      startingBalance = bigInt('0')
+    }
+    // Ask the faucet to send funds to the given address
+    const faucetURL = 'https://faucet.altnet.rippletest.net/accounts'
+    await axios.post(faucetURL, { destination: classicAddress })
+
+    // Wait for the faucet to fund our account or until timeout
+    // Waits one second checks if balance has changed
+    // If balance doesn't change it will attempt again until timeoutInSeconds
+    for (
+      let balanceCheckCounter = 0;
+      balanceCheckCounter < timeoutInSeconds;
+      balanceCheckCounter++
+    ) {
+      // Wait 1 second
+      await this.sleep(1000)
+
+      // Request our current balance
+      let currentBalance
+      try {
+        currentBalance = bigInt(await xrpClient.getBalance(address))
+      } catch {
+        currentBalance = bigInt('0')
+      }
+      // If our current balance has changed then return
+      if (startingBalance.notEquals(currentBalance)) {
+        return wallet
+      }
+
+      // In the future if we had a tx hash from the faucet
+      // We should check the status of the tx which would be more accurate
```

Comment:
is this worth tracking in a task?

Reply:
probably not - for our purposes here, we're generating a branch new wallet before requesting funds.  I think the only reason this would matter is if there were another client potentially manipulating the same address (or simultaneously sending funds to the same address) while we're tapping the faucet.  Just copied this comment from the wallet implementation of this same logic (where things might matter more)